packages/lexical-react/src/LexicalTablePlugin.ts

```diff
@@ -88,20 +88,18 @@ export function TablePlugin({
           return Math.max(curLength, row.length);
         }, 0);
         for (let i = 0; i < gridMap.length; ++i) {
-          const rowLength = gridMap[i].length;
+          const row = gridMap[i].filter((cell) => cell);
+          const rowLength = row.length;
```

Comment:
Since you're not using this array it would be more efficient to calculate it directly
```suggestion
          const rowLength = gridMap[i].reduce((acc, cell) => cell ? 1 + acc : acc, 0);
```